File: Services/CatalogService/Catalog.Application/Models/ProductDto.cs
namespace Catalog.Application.Models
{
    public class ProductDto
    {
        public Guid Id { get; init; }
        public required string Name { get; init; }
        public required string Description { get; init; }
        public decimal Price { get; init; }
        public int AvailableQty { get; init; }
        public Guid CategoryId { get; init; }
        public required IReadOnlyDictionary<string, string> Attributes { get; init; }
        public required IReadOnlyList<ProductImageDto> Images { get; init; }
    }
}

File: deliveryhub.client/src/models/catalog-service/ProductDto.ts
import type { UUIDTypes } from "uuid";
import type { BaseEntityIdentityGuid } from "../BaseEntityIdentityGuid";
import type { ProductImageDto } from "./ProductImageDto";

export interface ProductDto extends BaseEntityIdentityGuid {
    name: string;
    description: string;
    price: number;
    discount: number | 0;
    availableQty: number;
    categoryId: UUIDTypes;
    images: ProductImageDto[] | undefined;
    attributes: Record<string, string>;
}

export interface ProductSearchResultDto {
    products: ProductDto[];
    attributes: Record<string, string[]>;
}